pkg/virt-handler/virtwrap/manager.go

```diff
@@ -416,3 +427,125 @@ func (l *LibvirtDomainManager) KillVM(vm *v1.VM) error {
 	l.recorder.Event(vm, kubev1.EventTypeNormal, v1.Deleted.String(), "VM undefined")
 	return nil
 }
+
```

Comment:
lets get all of the VDSM specific code into its own file.  Split the comparable  libvirt specific code into its own file, and have an interface that both OvirtDomainManager and LibvirtDomain must comply with.  Ideally, these would be in separate modules, to enforce that they only use publicly sharable symbols.